juriscraper/scotus/scotus_docket_report_htm.py

```diff
@@ -0,0 +1,451 @@
+        :param current_party_name: The party name that this attorney belongs to.
+        :param parties_by_key: parties by type-party for grouping.
+        :return: None. The current_attorney is modified in place.
+        """
+        if current_attorney is None:
+            return None
+
+        lines = current_attorney.pop("_raw_lines", [])
+        email = current_attorney.pop("_email", None)
+
+        # Filter lines: remove lines with IDs like "#1098260" and inline email if present.
+        filtered_lines = []
+        for ln in lines:
+            if email is None:
+                m = self.EMAIL_RE.search(ln)
+                if m:
+                    email = m.group(0)
+                    continue
+            if self.ID_RE.search(ln):
+                continue
+            if ln:
+                filtered_lines.append(ln)
+
+        title, start_add_idx = self._parse_address_title(filtered_lines)
+        partial_address = self._parse_address_lines(
+            filtered_lines, start_add_idx
+        )
+
+        addr_lines = partial_address.address_lines
+        current_attorney.update(
+            {
+                "title": title,
+                "address": ", ".join(addr_lines) if addr_lines else None,
+                "city": partial_address.city,
+                "state": partial_address.state,
+                "zip": partial_address.zip_code,
+                "email": email,
+            }
+        )
+        type_key = (current_type or "Other", current_party_name)
+        parties_by_key[type_key].append(current_attorney)
+        return None
+
+    @property
+    def parties(self) -> list[dict[str, Any]]:
+        """Return parties grouped by type for HTM pages.
+
+        :return: List of dicts containing parties data.
+        """
+        if self.tree is None:
+            return []
+
+        # Locate the contacts table.
+        contacts_tbl = next(
+            iter(self.tree.xpath(self.CONTACTS_TABLE_XPATH)), None
+        )
+        if contacts_tbl is None:
+            return []
+
+        header_tr = next(iter(contacts_tbl.xpath(self.HEADER_ROW_XPATH)), None)
+        if header_tr is None:
+            return []
+
+        parties_by_key = defaultdict(list)
+        current_type = None
+        current_attorney = None
+
+        for tr in header_tr.itersiblings(tag="tr"):
+            # Match table header e.g: "Attorneys for Petitioner:"
+            header_text = self._clean_whitespace(
+                "".join(tr.xpath("./td[1]//b/text()"))
+            )
+            if header_text:
+                # Inside Petitioner, Respondent or Other parties.
+                parties_header = header_text.lower()
+                current_type = (
+                    "Petitioner"
+                    if "petitioner" in parties_header
+                    else "Respondent"
+                    if "respondent" in parties_header
+                    else "Other"
+                )
```

Comment:
This logic is similar to `_map_contacts_header_to_type`. We should refactor to reuse that method.